src/components/shoppingListItemCreateForm/shoppingListItemCreateForm.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  FormEventHandler,
+  useRef,
+  useState,
+  useEffect,
+  type CSSProperties,
+} from 'react'
+import classNames from 'classnames'
+import AnimateHeight from 'react-animate-height'
+import { RequestShoppingListItem } from '../../types/apiData'
+import { useShoppingListsContext, useColorScheme } from '../../hooks/contexts'
+import styles from './shoppingListItemCreateForm.module.css'
+
+interface CreateFormProps {
+  listId: number
+}
+
+const ShoppingListItemCreateForm = ({ listId }: CreateFormProps) => {
+  const { createShoppingListItem } = useShoppingListsContext()
+  const [expanded, setExpanded] = useState(false)
+  const formRef = useRef<HTMLFormElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const {
+    schemeColorDark,
+    hoverColorLight,
+    schemeColorLightest,
+    borderColor,
+    textColorSecondary,
+    textColorTertiary,
+  } = useColorScheme()
+
+  const colorVars = {
+    '--base-color': schemeColorDark,
+    '--hover-color': hoverColorLight,
+    '--body-color': schemeColorLightest,
+    '--border-color': borderColor,
+    '--text-color-main': textColorSecondary,
+    '--text-color-secondary': textColorTertiary,
+  } as CSSProperties
+
+  const toggleForm = () => {
+    setExpanded(!expanded)
+  }
+
+  const extractItem = (formData: FormData): RequestShoppingListItem => {
+    const attributes = Object.fromEntries(
+      Array.from(formData.entries()).filter(([_, value]) => !!value)
+    ) as Record<string, string>
+
+    const returnValue: RequestShoppingListItem = {
+      description: attributes.description,
+      quantity: Number(attributes.quantity),
+    }
+
+    if (attributes.unit_weight)
+      returnValue.unit_weight = Number(attributes.unit_weight)
+    if (attributes.notes) returnValue.notes = attributes.notes
+
+    return returnValue
+  }
+
+  const createItem: FormEventHandler = (e) => {
+    e.preventDefault()
+
+    if (!formRef.current) return
+
+    const formData = new FormData(formRef.current)
+    const attributes = extractItem(
+      formData
+    ) as unknown as RequestShoppingListItem
```

Comment:
This is redundant now that `extractData` actually returns a known-to-TypeScript `RequestShoppingListItem` object.